### PR TITLE
Add permission check methods

### PR DESCRIPTION
**Pull request by jjmason**
This PR also brings tests (integration) into the jenkins build, cleans up code considerably, and adds a few bonus methods.

Delivers [CON-3060].

### CODE REVIEW

File: conjur/config.py [hunk 1]
```diff
@@ -100,19 +99,26 @@ def set(self, key, value):
 
     pubkeys_url = _service_url('pubkeys', doc='URL for the pubkeys service')
 
-    verify_ssl = _setting('verify_ssl', True,
-                          "Allows SSL verification to be disabled (development \
-                          only!)")
-
```
Comment (kstutsman-conjur) [inline conjur/config.py]: Why was this block removed?

File: conjur/resource.py [hunk 2]
```diff
@@ -28,3 +32,74 @@ def __init__(self, api, kind, identifier):
...
+        '''
+        params = {
+            'permit': 'true',
+            'privilege': privilege,
+            'role': authzid(role)
+        }
+
+        self.api.post(self.url(), parmas=params)
+
+    def check_permission(self, privilege, role=None):
```
Comment (kstutsman-conjur) [inline conjur/resource.py]: Should this be called `permitted` or maybe have an alias of that since that's what the Ruby API uses?

File: conjur/config.py (hunk 1 above)
Comment (jjmason) [inline conjur/config.py]: We don't allow `verify_ssl` to be turned off now, because cert handling is so smooth with present day conjur (i.e. just run conjur init).   Removing this setting also reduces some confusion in the tests about what the arguments to `requests.request` should look like, and lets us unify verification stuff in the `verify()` method below.

File: conjur/resource.py [hunk 3]
```diff
@@ -28,3 +32,74 @@ def __init__(self, api, kind, identifier):
...
+
+        if role is None:
+            # Handle self role check
+            response = self.api.get(self.url(),
+                                    params={'check': 'true',
+                                            'privilege': privilege},
+                                    check_errors=False)
+            if response.status_code == 204:
+                return True
+            elif response.status_code in (404, 403, 409):
```
Comment (kstutsman-conjur) [inline conjur/resource.py]: I only see 404 and 403 handled by the Ruby API, but looks like the apiary docs mention 409. We should probably sort that out one way or another.

File: conjur/role.py [hunk 4]
```diff
@@ -27,20 +29,45 @@ def __init__(self, api, kind, identifier):
...
 
+    def is_permitted(self, resource, privilege):
+        params = {
+            'check': 'true',
+            'resource_id': authzid(resource, 'resource'),
+            'privilege': privilege
+        }
+        response = self.api.get(self._url(), params=params,
+                                check_errors=False)
+        if response.status_code < 300:
```
Comment (kstutsman-conjur) [inline conjur/role.py]: What's with the loose testing on the status code?

Comment (jjmason) [inline conjur/role.py]: Good point, I'll should make it demand `204`.

File: conjur/resource.py (hunk 3 above)
Comment (jjmason) [inline conjur/resource.py]: Yeah I was a little confused about that too.  I mean, I don't know why this would ever raise `409 conflict` since it's not modifying anything.   I suspect the Ruby API is doing the right thing and 409 is a copy/paste error in apiary.  I'll cut this back to 404 and 403.

File: conjur/util.py [hunk 5]
```diff
@@ -21,17 +21,17 @@
...
+    from urllib.parse import quote
 
 
 def urlescape(s):
     return quote(s, '')
 
 
 def authzid(obj, kind, with_account=True):
-    if isinstance(obj, basestring):
+    if isinstance(obj, (str, unicode)):  # noqa F821 (flake8 doesn't know about unicode)
```
Comment (kstutsman-conjur) [inline conjur/util.py]: I don't understand the reason for this change. If flake8 doesn't know about unicode, why wouldn't basicstring work?

File: conjur/resource.py (hunk 2 above)
Comment (jjmason) [inline conjur/resource.py]: Aliasing is super unpythonic (but very popular in ruby, even `Enumerable` and `Array` have multiple aliases for their methods).  `permitted` seems good, and worth doing before anybody starts writing code against this.

File: features/environment.py [hunk 6]
```diff
@@ -1,40 +1,61 @@
...
+
+def configure_logging():
+    logging.basicConfig(level=logging.DEBUG)
+
+
+def conjur_env(name, default=None):
+    print("conjur_env {} {}".format(name, default))
+    val = os.environ.get("CONJUR_{}".format(name.upper()), default)
+    print("got {}".format(val))
+    return val
```
Comment (kstutsman-conjur) [inline features/environment.py]: is this a debug function? It looks like it has debug prints in it

File: conjur/util.py (hunk 5 above)
Comment (jjmason) [inline conjur/util.py]: Neither works with flake8.  Not sure why the change.  It's equivalent I believe.

File: features/environment.py (hunk 6 above)
Comment (kstutsman-conjur) [inline features/environment.py]: Nevermind I see it's part of the cucumber configuration.

File: jenkins.sh [hunk 7]
```diff
@@ -1,13 +1,47 @@
...
-cp -R coverage.xml pytest.xml htmlcov pylint.out /artifacts/.
-COMMANDS
\ No newline at end of file
+export CONJUR_CERT_FILE=/certs/cuke-master.pem
+
+py.test --cov conjur --junitxml=pytest.xml --instafail
+coverage run --source='conjur/' -a -m behave
+coverage xml -o './coverage.xml'
+coverage html
+# pylint -f parseable conjur tests | tee pylint.out
```
Comment (kstutsman-conjur) [inline jenkins.sh]: Why's pylint commented out?

**Comment by kstutsman-conjur** [comment]
Code looks good to me. I know you're planning to make some changes from review so I won't merge it yet.

File: jenkins.sh (hunk 7 above)
Comment (jjmason) [inline jenkins.sh]: I was going to get it passing with this PR, but things were getting a little out of scope.
